src/services/DfusionService.ts

```diff
@@ -0,0 +1,243 @@
+        ])
+        const [sellToken, buyToken, validFrom, validUntil] = await Promise.all([
+          this._getToken(sellTokenAddress),
+          this._getToken(buyTokenAddress),
+          this._batchIdToDate(validFromBatchId),
+          this._batchIdToDate(validUntilBatchId)
+        ])
+
+        log.info(`New order in tx ${event.blockHash}:
+    - Owner: ${owner}
+    - Sell token: ${sellToken}
+    - Buy token: ${buyToken}
+    - Price: ${priceNumerator}/${priceDenominator} = ${priceNumerator.dividedBy(priceDenominator).toNumber()}
+    - Valid from: ${validFromBatchId}
+    - Valid until: ${validUntilBatchId}
+    - Block number: ${event.blockNumber}`)
+
+        params.onNewOrder({
+          owner,
+          sellToken,
+          buyToken,
+          priceNumerator,
+          priceDenominator,
+          validFromBatchId,
+          validUntilBatchId,
+          validFrom: validFrom as Date,
+          validUntil: validUntil as Date,
+          event
+        })
+      })
+      .on('changed', data => {
+        log.warn('Changed/Removed order: %o', data)
+      })
+      .on('error', (error: Error) => {
+        params.onError(error)
+      })
+  }
+
+  public async getAbout (): Promise<AboutDto> {
+    const [blockNumber, networkId, nodeInfo] = await Promise.all([
+      this._web3.eth.getBlockNumber(),
+      this._getNetworkId(),
+      this._web3.eth.getNodeInfo()
+    ])
+
+    return {
+      blockNumber,
+      networkId,
+      nodeInfo,
+      version: packageJson.version,
+      stablecoinConverterAddress: this._contract.options.address
+    }
+  }
+
+  private async _getNetworkId (): Promise<number> {
+    if (!this._networkId) {
+      this._networkId = await this._web3.eth.getChainId()
+    }
+
+    return this._networkId
+  }
+
+  private _getTokenAddress (id: string | number): Promise<string> {
+    return this._contract.methods.tokenIdToAddressMap(id).call()
+  }
+
+  private async _getToken (tokenAddress: string): Promise<TokenDto> {
+    let token = this._tokenCache[tokenAddress]
+
+    if (!token) {
+      const networkId = await this._getNetworkId()
+
+      const tokenContract = this._erc20Contract.clone()
+      tokenContract.options.address = tokenAddress
+
+      // Get basic data from the contract
+      const [symbol, decimals, name] = await Promise.all([
+        tokenContract.methods
+          .symbol()
+          .call()
+          .then(symbol => 'Maybe ' + symbol)
+          .catch(() => undefined),
+        tokenContract.methods
+          .decimals()
+          .call()
+          .then(parseInt)
+          .catch(() => 18),
+        tokenContract.methods
+          .name()
+          .call()
+          .then(name => 'Maybe ' + name)
+          .catch(() => undefined)
+      ])
+
+      const tokenJson = tokenList.find(token => token.addressByNetwork[networkId] === tokenAddress)
+      if (tokenJson) {
+        token = {
+          symbol,
+          name,
+          ...tokenJson,
```

Comment:
No need for condition. You can spread undefined and nothing will happen

Reply:
cool, didn't know that